app/models/embeds/pico_area.rb

```diff
@@ -0,0 +1,30 @@
+# This is a simple wrapper for Api::Area.
+# Sorry for the dutch. PICO is in dutch...
+
+class Embeds::PicoArea < SimpleDelegator
+  def self.find_by_area_code(area_code)
+    new Api::Area.find(area_code)
+  end
+
+  def area_name
+    return "Nederland" if type == :land
+    area.sub(/\A[^_]*/, "").humanize
+  end
+
+  def type
+    pattern_whitelist.each do |type, pattern|
+      return type if area.match?(pattern)
+    end
+    return :land
+  end
+
+  def to_json
```

Comment:
If this is send as a request response, you can instead define `as_json` returning a Ruby hash, and Rails will take care of the conversion and escaping the type and name.

```ruby
def as_json(*)
  { area_type: type, area_name: area_name }
end
```

Reply:
This is a bit of a hack. I'm using it to inject a javascript object literal into an onclick event handler(in embeds/picos/show.html). Its not meant to be used in many places. 

Since the class is just a decorator I didn't have problems with doing this. Are you of a different opinion?

Reply:
@antw I'd love to hear your opinion about it before I merge!

Reply:
> Are you of a different opinion?

It's not a strong opinion, no. :smiley: The only concern I'd have is if the area name has a `'` in it, the JSON will be invalid.

Reply:
> The only concern I'd have is if the area name has a ' in it, the JSON will be invalid.

... now that I write that, I'm realising that `to_json` is invalid JSON; the keys and values should both be surrounded with double quotes.